auditbeat/module/file_integrity/metricset_test.go

```diff
@@ -71,6 +71,10 @@ func TestActions(t *testing.T) {
 		t.Skip("Skipping flaky test: https://github.com/elastic/beats/issues/22518")
 	}
 
+	if runtime.GOOS == "darwin" {
+		t.Skip("Skipping flaky test: https://github.com/elastic/beats/issues/23965")
```

Comment:
There is already a skip on windows with a different github issue. The two issues seem to be duplicates.

Skimming the test code I noticed that a go-routine is started that is supposed to write the files. The test seems to be sensitive to timing + there is a chance of the test leaking a go-routine.

Does the test fail due to timing constraints and slow IO? What happens to the test if we add a `time.Sleep(2 * time.Minute)` in the helper go-routine (on sleep before and after each line in the go-routine)?

Reply:
@urso I wonder why we should add the sleep within the goroutine and not just right after it so as to give it time to do the IO ops and then let the main test to check for the events. I actually applied this on the latest commit(https://github.com/elastic/beats/pull/23966/commits/b10fe6cf95f19881705c8ac0899e4f1aad9db3bf) and seems to work.

Reply:
we don't know how long we have to wait for the IO. The `time.Sleep` was for you to test if the test is too time sensitive... as evidence, not to fix the test.

e.g.
```
	go func() {
        time.Sleep(2 * time.Minute)
		ioutil.WriteFile(createdFilepath, []byte("hello world"), 0600)
        time.Sleep(2 * time.Minute)
		ioutil.WriteFile(updatedFilepath, []byte("hello world"), 0600)
        time.Sleep(2 * time.Minute)
	}()
```

If the test fails with these timeouts, it is a good indicator that the test is too sensitive to IO scheduler timings (which can easily block write operations for a minute if the system is overloaed).

If adding a `time.Sleep` after the go-routine really stabilizes the test the question is: why do we need to have the go-routine in the first place?

@andrewkroh Any idea if the go-routine is required for the test?

Reply:
I don't see why there is a goroutine with only `WriteFile`s. I expected there to be a `time.Sleep` in the goroutine followed by the `WriteFile` similar to https://github.com/elastic/beats/blame/b10fe6cf95f19881705c8ac0899e4f1aad9db3bf/auditbeat/module/file_integrity/metricset_test.go#L49-L53. I think it was a basic way to schedule some activity to happen after the  MetricSet is running.

Reply:
Thank you. In that we should do the WriteFile without the go-routine.